docs/extended_api/synchronization_primitives/barrier/wait_parity.md

```diff
@@ -0,0 +1,45 @@
+---
+grand_parent: Extended API
+parent: Barriers
+---
+
+# `cuda::barrier::wait_parity` and `cuda::barrier::try_wait_parity`
+
+Defined in header `<cuda/std/barrier>`:
+
+```cuda
+__host__ __device__ void cuda::std::barrier::wait_parity(bool phase);
+__host__ __device__ bool cuda::std::barrier::try_wait_parity(bool phase);
+```
+
+`barrier::wait_parity` stalls execution while the barrier is not at the specified parity.
+`barrier::try_wait_parity` returns true if the parity of the barrier matches the given parity.
+
+## Return Value
+
+`barrier::try_wait_parity` returns a boolean representing whether the barrier is at the given phase
+
+<!-- TODO: Create an example when trunk is live on godbolt
+## Example
+
+```cuda
+#include <cuda/barrier>
+
+__global__ void example_kernel(cuda::barrier<cuda::thread_scope_block>& bar) {
+  bar.wait_parity(false);
+}
+```
```

Comment:
It would be nice to have an example that shows how to use this to accomplish something minimally useful.

In particular, it might not be clear to a user, e.g., what the initial parity of a barrier is, how to get the parity of a barrier, etc.